src/cogent3/app/align.py

```diff
@@ -553,6 +559,65 @@ def main(self, seqs: UnalignedSeqsType) -> T:
         return result
 
 
+@define_app
+class smith_waterman:
+    """Local alignment of two sequences using the Smith-Waterman algorithm
+
+    Notes
+    -----
+    It records the score of the alignment (in addition to all other the
+    parameter values) as "sw_score" in ``alignment.info['align_params']``.
+    """
+
+    def __init__(
+        self,
+        score_matrix=None,
+        insertion_penalty=20,
+        extension_penalty=2,
+        moltype="dna",
+    ):
+        """
+        Parameters
+        ----------
+        score_matrix : dict
+            scoring dict, defaults to `make_dna_scoring_dict(10, -1, -8)` for DNA
+            and `make_generic_scoring_dict(10, moltype)` for other moltype.
+        insertion_penalty : int
+            penalty for gap insertion
+        extension_penalty : int
+            penalty for gap extension
+        moltype
+            molecular type of sequences
+        """
+        moltype = get_moltype(moltype)
+        self.score_matrix = score_matrix or (
+            make_dna_scoring_dict(10, -1, -8)
+            if moltype.label == "dna"
+            else make_generic_scoring_dict(10, moltype)
+        )
+        self.insertion_penalty = insertion_penalty
+        self.extension_penalty = extension_penalty
+
+    def main(self, seq1: SeqType, seq2: SeqType) -> AlignedSeqsType:
```

Comment:
this won't work with the standard app approach, a single argument is required. Change to
```python
def main(self, seqs: UnalignedSeqsType) -> AlignedSeqsType:
      if seqs.num_seqs > 2:
        return NotCompleted("ERROR", ..., message="maximum number of two seqs per collection", ...)
     seq1, seq2 = seqs.seqs
   ```